src/dex-ui-components/SwapTokens/types.ts

```diff
@@ -1,24 +1,50 @@
-export interface SwapTokensState {
-  tokenToTrade: {
-    symbol: string | undefined;
-    amount: number;
-    displayAmount: string;
-    balance: number | undefined;
-    poolLiquidity: number | undefined;
-  };
-  tokenToReceive: {
-    symbol: string | undefined;
-    amount: number;
-    displayAmount: string;
-    balance: number | undefined;
-    poolLiquidity: number | undefined;
+export interface TokenState {
+  symbol: string | undefined;
+  amount: number;
+  displayAmount: string;
+  balance: number | undefined;
+  poolLiquidity: number | undefined;
+  tokenName: string | undefined;
+  totalSupply: Long | null;
```

Comment:
What's the reasoning behind using the `Long` type instead of the `BigNumber` library for these values?

Reply:
The totalSupply from TokenData returns the `Long | null` type to match it I kept it this way